src/main/java/mekanism/common/content/gear/mekatool/ModuleVeinMiningUnit.java

```diff
@@ -49,30 +51,40 @@ public int getExcavationRange() {
         return excavationRange.get().getRange();
     }
 
-    public static Set<BlockPos> findPositions(BlockState state, BlockPos location, Level world, int maxRange) {
+    public static Set<BlockPos> findPositions(Level world, Set<BlockPos> initial, boolean extended, int maxRange) {
         Set<BlockPos> found = new LinkedHashSet<>();
         Set<BlockPos> openSet = new LinkedHashSet<>();
-        openSet.add(location);
-        Block startBlock = state.getBlock();
-        int maxCount = MekanismConfig.gear.disassemblerMiningCount.get() - 1;
+        openSet.addAll(initial);
+        int maxCount = (MekanismConfig.gear.disassemblerMiningCount.get() - 1) * initial.size();
+        AtomicInteger next = new AtomicInteger(openSet.size());
+        AtomicInteger dist = new AtomicInteger();
         while (!openSet.isEmpty()) {
             BlockPos blockPos = openSet.iterator().next();
             found.add(blockPos);
             openSet.remove(blockPos);
             if (found.size() > maxCount) {
                 return found;
+            } else if (found.size() == next.get()){
+                dist.getAndIncrement();
+                next.addAndGet(openSet.size());
             }
-            for (BlockPos pos : BlockPos.betweenClosed(blockPos.offset(-1, -1, -1), blockPos.offset(1, 1, 1))) {
-                //We can check contains as mutable
-                if (!found.contains(pos) && (maxRange == -1 || WorldUtils.distanceBetween(location, pos) <= maxRange)) {
-                    Optional<BlockState> blockState = WorldUtils.getBlockState(world, pos);
-                    if (blockState.isPresent() && startBlock == blockState.get().getBlock()) {
-                        //Make sure to add it as immutable
-                        if (openSet.add(pos.immutable())) {
-                            //Note: We do this for all blocks we find/attempt to mine, not just ones we do mine, as it is a bit simpler
-                            // and also represents those blocks getting checked by the vein mining for potentially being able to be mined
-                            Mekanism.packetHandler().sendToAllTracking(new PacketLightningRender(LightningPreset.TOOL_AOE, Objects.hash(blockPos, pos),
-                                  Vec3.atCenterOf(blockPos), Vec3.atCenterOf(pos), 10), world, blockPos);
+            BlockState state = world.getBlockState(blockPos);
+            boolean isOre = state.is(MekanismTags.Blocks.ATOMIC_DISASSEMBLER_ORE);
+            //If it is extended or should be treated as an ore
+            if (isOre || extended) {
```

Comment:
This feels like it is wrong as previously it was validating if it is the same block as the original state whereas now if you mine a log next to say a piece of coal ore I am pretty sure it will mine both of them.

Reply:
That check is just to see if it can vein mine and whether to check the adjacent blocks, what prevents logs vein mining coal is the block check lower down.

Reply:
Ah whoops when I was looking at the changes it looked like the same block check had been entirely removed.